torch/__init__.py

```diff
@@ -750,17 +750,17 @@ def set_default_tensor_type(t):
 def set_default_dtype(d):
     r"""
 
-    Sets the default floating point dtype to :attr:`d`. Supports torch.float32
-    and torch.float64 as inputs. Other dtypes may be accepted without complaint
-    but are not supported and are unlikely to work as expected.
+    Sets the default floating point dtype to :attr:`d`. Supports floating point dtype
+    as inputs. Other dtypes will cause torch to raise an exception.
 
     When PyTorch is initialized its default floating point dtype is torch.float32,
-    and the intent of set_default_dtype(torch.float64) is to facilitate NumPy-like
+    and the intent of set_default_dtype(...) is to facilitate NumPy-like
```

Comment:
The "NumPy-like type inference" in this sentence refers specifically to `float64` which is the default floating point type in NumPy.